src/import.js

```diff
@@ -250,6 +250,69 @@ padlock.import = (function(crypto, DisposableSource, Papa) {
         return data.split("\n")[0] == "url,username,password,extra,name,grouping,fav";
     }
 
+    /**
+     * Checks if a given string represents a KeePassX exported XML file
+     */
+    function isKeePassXML(xmldata) {
+        return xmldata.split("\n")[0] == "<!DOCTYPE KEEPASSX_DATABASE>";
+    }
+
+    /**
+     * Imports data from the XML file exported by KeePassX
+     */
+     function importKeePassXML(xmldata) {
+
+         // parse the xmldata string into actual XML format
+         parser = new DOMParser();
+         xmlDoc = parser.parseFromString(xmldata,"text/xml");
+
+         // initialize list of all records in this XML file
+         var records = [];
+
+         // get all the "groups" (aka categories) from xmlDoc and loop through them
+         var groups = xmlDoc.getElementsByTagName("group");
+         for (group = 0; group < groups.length; group++) {
+             var category = groups[group].getElementsByTagName("title")[0];
+             var category = category.childNodes[0].nodeValue;
```

Comment:
Nevermind... what you're doing is right. However, a more concise way of writing it would be
```js
var titleNode = groups[group].querySelector("title"); // Will only return the first title node
var category = titleNode ? titleNode.innerHTML : "";
```
Actually, not much more concise but a little safer and easier to read imo.